src/SonarScanner.MSBuild.Shim/AdditionalFilesService.cs

```diff
@@ -74,20 +74,17 @@ public AdditionalFiles AdditionalFiles(AnalysisConfig analysisConfig, DirectoryI
             : PartitionAdditionalFiles(allFiles, analysisConfig);
     }
 
-    private List<string> GetAllFiles(IEnumerable<string> extensions, DirectoryInfo projectBaseDir)
-    {
-        var pattern = string.Join("|", extensions.Select(x => x.TrimStart('.') + "$").Distinct());
-        var regex = new Regex(pattern, RegexOptions.Compiled | RegexOptions.IgnoreCase, RegexTimeout);
-        return directoryWrapper
+    private List<FileInfo> GetAllFiles(IEnumerable<string> extensions, DirectoryInfo projectBaseDir) =>
+        directoryWrapper
             .EnumerateFiles(projectBaseDir.FullName, "*", SearchOption.AllDirectories)
-            .Where(regex.SafeIsMatch)
+            .Where(x => extensions.Any(e => x.EndsWith(e) && !x.EndsWith($"{Path.DirectorySeparatorChar}{e}")))
+            .Select(x => new FileInfo(x))
```

Comment:
You should do the `new FileInfo` before the Where. It should simplify the code because you can use `Name` for the second check (`!x.Name.Equals(e)`)